src/index.ts

```diff
@@ -116,21 +130,18 @@ const createProxyHandler = <T extends object>(origObj: T, frozen: boolean) => {
         recordObjectAsUsed();
         return true;
       }
-      // LIMITATION: We simply record the same as `get`.
-      // This means { a: {} } and { a: {} } is detected as changed,
-      // if `'a' in obj` is handled.
-      recordUsage(key);
+      getUsed()?.[HAS_KEY_PROPERTY].add(key);
       return key in target;
     },
     getOwnPropertyDescriptor(target, key) {
-      // LIMITATION: We simply record the same as `get`.
-      // This means { a: {} } and { a: {} } is detected as changed,
-      // if `obj.getOwnPropertyDescriptor('a'))` is handled.
-      recordUsage(key, true);
+      const used = getUsed();
+      if (used && !used[HAS_OWN_KEY_PROPERTY].has(OWN_KEYS_SYMBOL)) {
```

Comment:
Why do we need this condition?

Reply:
Great question! I just wanted to preserve `skipWithOwnKeys` behavior. If we detected that user has called `Object.keys()` on our proxy - we will check that all own keys didn't change so we don't have to track individual keys anymore.